src/domain/valueObject/unixFilePermissions.go

```diff
@@ -0,0 +1,65 @@
+package valueObject
```

Comment:
Você vai precisar refazer esse VO aqui, a lógica de construção não tá legal. Você possui o InterfaceToUint, logo bastava usar ele logo no construtor principal.

Permissões podem ter 3 ou 4 digits, então você não necessariamente precisa forçar o modelo de 4 dígitos, pois na infra você vai conseguir setar permissões com 3 ou 4 dígitos. 

Caso você seja obrigado a usar o modelo de 4 dígitos em algum lugar no código, crie um getter pra isso, algo como GetWith4Digits e lá você pode add o zero a esquerda caso o número seja 3 chars.